test/integration/load_node_test.cpp

```diff
@@ -185,6 +185,20 @@ TEST(LoadNodeTest, DereferenceIteratorError) {
   EXPECT_THROW(node.begin()->begin()->Type(), InvalidNode);
 }
 
+TEST(NodeTest, MergeKeyScalarSupport) {
```

Comment:
Like, the main dictionary has the key "foo", and so does the merge entry - the test should confirm that "foo" refers to the main dictionary.